include/onnxruntime/core/graph/basic_types.h

```diff
@@ -29,6 +29,8 @@ using ProviderType = const std::string&;
 using NodeAttributes = std::unordered_map<std::string, ONNX_NAMESPACE::AttributeProto>;
 class IOnnxRuntimeOpSchemaCollection;
 using IOnnxRuntimeOpSchemaCollectionPtr = std::shared_ptr<IOnnxRuntimeOpSchemaCollection>;
+class SchemaRegistryManager;
+using SchemaRegistryManagerPtr = std::shared_ptr<SchemaRegistryManager>;
```

Comment:
Because like IOnnxRuntimeOpSchemaCollectionPtr in present master, SchemaRegistryManager indeed has to be shared by model, graph and sub-graphs.

Reply:
Can you fix the IOnnxRuntimeOpSchemaCollection one too? Ptr is something we removed in the C APIs because it confused everyone.

ObjectPtr *p;

What is it? A **? A unique_ptr? A shared_ptr? It's easy to use incorrectly

Reply:
Ambiguity removed.
Thanks!